{"ast":null,"code":"const maxRetry = 3;\n\nclass GitError extends Error {\n  shouldRetry() {\n    return false;\n  }\n\n}\n\nclass GitConnectionError extends GitError {\n  constructor(message) {\n    super('A git connection error occurred');\n  }\n\n  shouldRetry(number) {\n    return number < maxRetry;\n  }\n\n}\n\nclass GitPathspecError extends GitError {\n  constructor(message) {\n    super('The git reference could not be found');\n  }\n\n}\n\nclass GitUnknownError extends GitError {\n  constructor(message) {\n    super('An unknown git error occurred');\n  }\n\n}\n\nmodule.exports = {\n  GitConnectionError,\n  GitPathspecError,\n  GitUnknownError\n};","map":null,"metadata":{},"sourceType":"script"}